packages/java/parser-jvm-core/src/main/java/dev/hilla/parser/models/ParameterizedModel.java

```diff
@@ -0,0 +1,12 @@
+package dev.hilla.parser.models;
+
+import java.util.List;
+import java.util.stream.Stream;
+
+public interface ParameterizedModel {
+    List<TypeParameterModel> getTypeParameters();
+
+    default Stream<TypeParameterModel> getTypeParametersStream() {
+        return getTypeParameters().stream();
+    }
```

Comment:
This method has no added value and in general it is better not to return `Stream`s unless you are transforming them or unless that really allows you to skip the creation of an intermediate collection.

Reply:
Well, it streamlines final code, but overcomplicates the API, so it is debatable. Personally, I'm not fond of it. And in this very case it is not even used like that.

Reply:
Well, sounds reasonable. Though, I guess, if we decide to remove these methods, we should remove all at once from all the models.